docker/bin/zookeeperStart.sh

```diff
@@ -92,13 +92,6 @@ if [[ "$WRITE_CONFIGURATION" == true ]]; then
     echo $ZKCONFIG
     echo $MYID > $MYID_FILE
     echo "server.${MYID}=${ZKCONFIG}" > $DYNCONFIG
-  else
```

Comment:
This code block is executed only when "$WRITE_CONFIGURATION" == true,
which means it is executed for any zookeeper server with ID >1 when it is coming up for the very first time.
For the usecase mentioned in the issue, removing this code block would also lead to a Crashloopbackoff state on the ZK pod, since with $REGISTER_NODE==false, the "DYNCONFIG" (_points to $DATA_DIR/zoo.cfg.dynamic_) will not exist and so the script would exit with status 1.
Has this change been tested with all possible cases of zookeeper server start, restarts, scale up/down, upgrade and does it work through all of those?

Reply:
> This code block is executed only when "$WRITE_CONFIGURATION" == true,
which means it is executed for any zookeeper server with ID >1 when it is coming up for the very first time.

WRITE_CONFIGURATION is used for ID=1 only IIUC.
All others are peers are configured (i.e $DYNCONFIG  created) in:

https://github.com/pravega/zookeeper-operator/blob/20106e883c62d143eaa0c1c26fd7c59f0328dcc8/docker/bin/zookeeperStart.sh#L117-L125


> For the usecase mentioned in the issue, removing this code block would also lead to a Crashloopbackoff state on the ZK pod, since with $REGISTER_NODE==false, the "DYNCONFIG" (points to $DATA_DIR/zoo.cfg.dynamic) will not exist and so the script would exit with status 1.


The pod *won't* enter CrashhLoopBackOff:
If `$REGISTER_NODE==false` it means that either the headless service is not available or the node already registered. In the first case - when headless service is not available it will exit 1 and retry, awaiting for headless service to come up. This is a good behaviour IMO.

But here is the thing - if you keep the `else` branch here, when headless service is not available then 
```sh
CONFIG=`java -Dlog4j.configuration=file:"$LOG4J_CONF" -jar /root/zu.jar get-all $ZKURL`
echo Writing configuration gleaned from zookeeper ensemble
echo "$CONFIG" | grep -v "^version="> $DYNCONFIG
```
will write an empty $DYNCONFIG file, the  `zkServer.sh` peer starts fails (invalid dyn file) and even worse on next pod restart the $DYNCONFIG file is available (albeit empty) and 
https://github.com/pravega/zookeeper-operator/blob/20106e883c62d143eaa0c1c26fd7c59f0328dcc8/docker/bin/zookeeperStart.sh#L77-L79

will set `WRITE_CONFIGURATION` to `false` so DYNCONFIG is forever empty and invalid.

The whole logic here is convoluted in my opinion - I think it can be simplified but in our setup this fixed the issues.